pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -75,6 +74,14 @@
  */
 public class RealtimeSegmentAssignment extends BaseSegmentAssignment {
 
+  @Override
+  protected void setSegmentAssignmentStrategyMap(
```

Comment:
can consider moving this implementation up to be implemented in `BaseSegmentAssignment` since `OfflineSegmentAssignment` also shares same implementation and looks like `SegmentAssignmentStrategyFactory.getSegmentAssignmentStrategy` can handle the specifics.